tezos/new_context/src/tezedge_context.rs

```diff
@@ -625,26 +697,42 @@ impl ShellContextApi for TezedgeContext {
         message: String,
         date: i64,
     ) -> Result<ContextHash, ContextError> {
-        // FIXME: does more work than needed, just calculate the hash, no batch
         let date: u64 = date.try_into()?;
-        let (commit_hash, _batch, _referenced_older_entries) =
-            self.tree
-                .prepare_commit(date, author, message, self.parent_commit_hash)?;
-        let commit_hash = ContextHash::try_from(&commit_hash[..])?;
+        let mut repository = self.index.repository.write()?;
+
+        let (commit_hash_id, batch, _referenced_older_entries) = self.tree.prepare_commit(
+            date,
+            author,
+            message,
+            self.parent_commit_hash,
+            &mut *repository,
+        )?;
+
+        repository.write_batch(batch)?;
```

Comment:
This `hash` method shouldn't be writing anything to the repository, all it needs to do is to produce a `Commit` to calculate its hash

Reply:
If this is done so that you can get the hash, what can be done instead is to create a `Commit` manually (in the same way that `prepare_commit` does and pass that to `hash_commit`, or maybe even better yet, add a `hash_commit` to `WorkingTree` that just produces the hash.

Reply:
Ok, I see that `hash_commit` returns `HashId`, so it will have to be split into two functions, one that does the actual work (let's say `hash_commit_into`) and writes the resulting hash into a slice passed as an argument, and then `hash_commit` can use that. And `WorkingTree.hash_commit(author, message, date)` will just produce the hash directly.